proposals/lip-0052.md

```diff
@@ -362,712 +379,1832 @@ The module provides the following commands to modify the NFT store.
 ```
 
+The [transferInternal](#transferinternal) function transfers the ownership of the NFT within the chain.
+
 #### Cross-chain NFT Transfer
 
 Transactions executing this command have:
 
-* `moduleID = NFT_MODULE_ID`
-* `commandID  = COMMAND_ID_CROSS_CHAIN_TRANSFER`
+- `module = MODULE_NAME_NFT`
+- `command = COMMAND_NAME_CROSS_CHAIN_TRANSFER`
 
 ##### Parameters Schema
 
-The `params` property of a cross-chain NFT transfer transaction follows the schema `crossChainTransferParams`.
+The `params` property of a cross-chain NFT transfer transaction follows the `crossChainTransferParamsSchema` schema.
 
 ```java
-crossChainTransferParams = {
+crossChainTransferParamsSchema = {
     "type": "object",
     "required": [
-        "nftID",   
+        "nftID",
         "receivingChainID",
         "recipientAddress",
+        "data",
         "messageFee"
     ],
     "properties": {
         "nftID": {
-            "type": object,
+            "dataType": "bytes",
+            "length": NFT_ID_LENGTH,
             "fieldNumber": 1,
-            "required": [
-                "chainID",
-                "collection",
-                "index"
-            ],
-            "properties": {
-                "chainID": {
-                    "dataType": "uint32",
-                    "fieldNumber": 1
-                },
-                "collection": {
-                    "dataType": "uint32",
-                    "fieldNumber": 2
-                },
-                "index": {
-                    "dataType": "uint64",
-                    "fieldNumber": 3
-                }
-            }
         },
         "receivingChainID": {
-            "dataType": "uint32",
+            "dataType": "bytes",
+            "length": CHAIN_ID_LENGTH,
             "fieldNumber": 2
         },
         "recipientAddress": {
             "dataType": "bytes",
+            "length": ADDRESS_LENGTH,
             "fieldNumber": 3
         },
+        "data": {
+            "dataType": "string",
+            "maxLength": MAX_DATA_LENGTH,
+            "fieldNumber": 4
+        }
         "messageFee": {
             "dataType": "uint64",
-            "fieldNumber": 4
+            "fieldNumber": 5
         }
     }
 }
 ```
 
-##### Parameters Validity
+##### Verification
 
-The `params` property of a cross-chain NFT transfer transaction is valid if:
+```python
+def verify(trs: Transaction) -> None:
 
-* `recipientAddress` is a byte array of length `ADDRESS_LENGTH`,
-* `tokenID.chainID` is either `CHAIN_ID_ALIAS_NATIVE` or `receivingChainID`,
+    senderAddress = sha256(trs.senderPublicKey)[:ADDRESS_LENGTH]
+    nftID = trs.params.nftID
+    receivingChainID = trs.params.receivingChainID
+    recipientAddress = trs.params.recipientAddress
+    data = trs.params.data
+    messageFee = trs.params.messageFee
 
-##### Execution
+    if NFTStore(nftID) does not exist:
+        raise Exception("NFT store entry does not exist.")
+
+    if len(getNFTOwner(nftID)) == CHAIN_ID_LENGTH:
```

Comment:
As before, I think this check is superfluous given the following one on the senderAddress. Maybe it is fine to keep it to give a more granular error message? In that case you should also add it to the normal transfer function.